### DISCUSSION TITLE
How to retrieve user permissions (claims) in version 2.1.0. since it's removed from the user principal

### DISCUSSION THREAD

**Discussion by Kees-Schouten** [OrchardCMS/OrchardCore#17052] [General]
Since Orchard 2.1.0 the user principal permission claims are removed. The user role claims are still available though.
My app authenticates using OpenId.
What is the best way to retrieve those. I need a list of the user permissions in our frontend application.

I have found some code which does something like that, see:
OrchardCore.Roles.Controllers.AdminController.GetInstalledPermissionsAsync

I'm hoping there's something out of the box.

**Comment by Piedone** [comment]
I'm afraid what's in `GetInstalledPermissionsAsync` is pretty much what you have to do. Are you running decoupled or headless?

**Reply by KeesSchoutenEasySystems** [reply]
Ok thanks! I'm running sort of headless. I've created an data abstraction layer for querying data from Orchard and use basic Orchard features like, media, users, openid, etc. 

Do you know, how Orchard did retrieved the permissions claims before version version 2.1.0?. I tried to find it in git history but couldn't find it.

**Reply by Piedone** [reply]
@MikeAlhayek I believe you changed something in this area, but can't find it now, could you please chime in?

**Reply by KeesSchoutenEasySystems** [reply]
I've managed to implement it using GetInstalledPermissionsAsync. Thanks

**Reply by Piedone** [reply]
Great!
